editor/components/block-list/index.js

```diff
@@ -3,294 +3,44 @@
-		// Cache an array of the Y coördinates for use in `onPointerMove`.
-		// Sort the coördinates, as `this.nodes` will not necessarily reflect
-		// the current block sequence.
-		this.coordMapKeys = sortBy( Object.values( uidToCoordMap ) );
-		this.selectionAtStart = uid;
-
-		window.addEventListener( 'mousemove', this.onPointerMove );
-		// Capture scroll on all elements.
-		window.addEventListener( 'scroll', this.onScroll, true );
-		window.addEventListener( 'mouseup', this.onSelectionEnd );
+function BlockList( { blocks, layouts = {}, rootUID } ) {
+	const isGroupedByLayout = Array.isArray( layouts );
+	if ( ! isGroupedByLayout ) {
+		layouts = [ { name: undefined } ];
 	}
 
-	onSelectionChange( uid ) {
-		const { onMultiSelect, selectionStart, selectionEnd } = this.props;
-		const { selectionAtStart } = this;
-		const isAtStart = selectionAtStart === uid;
-
-		if ( ! selectionAtStart || ! this.props.isSelectionEnabled ) {
-			return;
-		}
-
-		if ( isAtStart && selectionStart ) {
-			onMultiSelect( null, null );
-		}
-
-		if ( ! isAtStart && selectionEnd !== uid ) {
-			onMultiSelect( selectionAtStart, uid );
-		}
-	}
-
-	/**
-	 * Handles a mouseup event to end the current cursor multi-selection.
-	 *
-	 * @returns {void}
-	 */
-	onSelectionEnd() {
-		// Cancel throttled calls.
-		this.onPointerMove.cancel();
-
-		delete this.coordMap;
-		delete this.coordMapKeys;
-		delete this.selectionAtStart;
-
-		window.removeEventListener( 'mousemove', this.onPointerMove );
-		window.removeEventListener( 'scroll', this.onScroll, true );
-		window.removeEventListener( 'mouseup', this.onSelectionEnd );
-
-		// We may or may not be in a multi-selection when mouseup occurs (e.g.
-		// an in-place mouse click), so only trigger stop if multi-selecting.
-		if ( this.props.isMultiSelecting ) {
-			this.props.onStopMultiSelect();
-		}
-	}
-
-	onShiftSelection( uid ) {
-		if ( ! this.props.isSelectionEnabled ) {
-			return;
-		}
-
-		const { selectedBlock, selectionStart, onMultiSelect, onSelect } = this.props;
-
-		if ( selectedBlock ) {
-			onMultiSelect( selectedBlock.uid, uid );
-		} else if ( selectionStart ) {
-			onMultiSelect( selectionStart, uid );
-		} else {
-			onSelect( uid );
-		}
-	}
-
-	render() {
-		const { blocks, showContextualToolbar } = this.props;
+	return map( layouts, ( layout ) => {
+		const layoutBlocks = isGroupedByLayout ?
+			filter( blocks, ( block ) => (
+				get( block, [ 'attributes', 'layout' ] ) === layout.name
+			) ) :
+			blocks;
 
 		return (
-			<div>
-				{ !! blocks.length && <BlockListSiblingInserter /> }
-				{ map( blocks, ( uid ) => (
-					<BlockListBlock
-						key={ 'block-' + uid }
-						uid={ uid }
-						blockRef={ this.setBlockRef }
-						onSelectionStart={ this.onSelectionStart }
-						onShiftSelection={ this.onShiftSelection }
-						showContextualToolbar={ showContextualToolbar }
-					/>
-				) ) }
-			</div>
+			<BlockListLayout
+				key={ layout.name || 'default' }
+				layout={ layout.name }
```

Comment:
@aduth This is just a minor thing you may already be aware of.

![2018-01-21_02-21-03](https://user-images.githubusercontent.com/1563559/35193628-d36850d0-fe51-11e7-80bc-c63f4a75c910.png)